src/Ng/ElfieFromCatalogCollector.cs

```diff
@@ -135,20 +135,30 @@ public ElfieFromCatalogCollector(Uri index, Storage storage, int maxThreads, Fun
         /// <param name="catalogItem">The catalog item to process.</param>
         async Task ProcessPackageDetailsAsync(CatalogItem catalogItem, CancellationToken cancellationToken)
         {
-            Trace.TraceInformation("#StartActivity ProcessPackageDetailsAsync " + catalogItem.Id + " " + catalogItem.PackageVersion);
+            Trace.TraceInformation("#StartActivity ProcessPackageDetailsAsync " + catalogItem.PackageId + " " + catalogItem.PackageVersion);
 
-            // Download the registration json file.  
-            // The registration file will tell if the package is listed as-well-as provide the package download URL.  
-            RegistrationItem registrationItem = catalogItem.GetRegistrationItem(this._nugetServiceUrls);
+            // Do not process prerelease packages
+            if (catalogItem.IsPrerelease)
+            {
+                Trace.TraceInformation("Skipping prerelease package");
+                return;
+            }
 
-            // We only need to process listed packages  
-            if (registrationItem.Listed)
+            RegistrationIndexPackage latestStablePackage = catalogItem.GetLatestStableVersion(this._nugetServiceUrls);
+            if (latestStablePackage == null)
+            {
+                Trace.TraceInformation("Skipping package without a released version");
+                return;
+            }
+            else if (!latestStablePackage.CatalogEntry.PackageVersion.Equals(catalogItem.PackageVersion))
             {
-                Uri packageResourceUri = await this.DownloadPackageAsync(catalogItem, registrationItem.PackageContent, cancellationToken);
+                Trace.TraceInformation("Skipping historical package");
```

Comment:
A released (listed) package that is not the latest version. I didn't know what to call it. Any suggestions?